contents/blog/eu-cloud-hosting.md

```diff
@@ -0,0 +1,89 @@
+
+Next was choosing a domain. Our team initially thought to use `posthog.eu` but that raised concerns about cross-domain tracking and SEO. It also potentially limited future expansion. Another proposal was to use a longer domain like `app.eu-central-1.posthog.com`but this was too long. Just keeping `app.posthog.com` and redirecting traffic was also an option, but there was complexity around accounts in multiple organizations and sign in.
+
+Eventually, we settled on `eu.posthog.com`. It is short, memorable, and allows for other regions to follow in the future.
+
+> Have opinions on where we should host in the future? We’d love to hear them. Let us know in our [Community Slack](/slack).
+
+## Setting up key infrastructure
+
+As mentioned before, because the Infrastructure Team had done a lot of work at improving the deployment, scaling, and managing of infrastructure recently, the process of setting up key infrastructure was easier. Structuring PostHog to be easy to deploy and scale is important because, PostHog is designed to allow for self-hosting.
+
+As work was starting, Guido from our team said: 
+
+> "Spinning up PostHog Cloud EU feels (at least to me) a bit like playing lego. It’s incredibly satisfying to see the effort we put in the last months to designing and building those pieces is finally paying off."
+
+One of the first steps was making sure to first split the production integration and deployment pipeline from US Cloud. Next was setting up and configuring the key infrastructure for PostHog such as ClickHouse, Kafka (with AWS MSK), Redis, ZooKeeper, Kubernetes (AWS EKS), and more. See our docs on [how PostHog works](/docs/how-posthog-works) for more details.
+
+They made some slight changes, moving the database from Heroku to AWS RDS (Relational Database). This is better future-proofing, as Heroku has scaling limits and is difficult to migrate from. Other changes led to upgrades, like more shards in our ClickHouse cluster, allowing for easier scaling in the future.
+
+## Dealing with infrastructure challenges
+
+The first key challenge was separating correctly from US Cloud to ensure changes wouldn’t break both. EU Cloud needed a recreated CI/CD pipeline separate from the US Cloud one. For example, moving from Heroku to AWS RDS required changes to the deployment pipeline.
+
+A stranger challenge was getting SSL certificates. CloudFront only supports certificates from `us-east-1` as the control plane is regional despite being a global service ([more details](https://aws.amazon.com/premiumsupport/knowledge-center/migrate-ssl-cert-us-east/)). This required some reconfiguring, but once we realized it, it was a quick change.
+
+Billing required a significant amount of work to get working on EU Cloud because it required different configurations for EU vs US. The test, production, EU, and US instances needed extra environment variables to configure. Settings were tweaked multiple times before it worked
```

Comment:
```suggestion
Billing required a significant amount of work to get working on EU Cloud because it needed different configurations for EU vs US. The test, production, EU, and US instances needed extra environment variables to configure and we tweaked settings multiple times before it worked. 
```

Try to avoid using the same verb repeatedly in a single sentence. 
Missing full stop on the end
Passive voice in last sentence.